src/strands/models/bedrock.py

```diff
@@ -327,7 +318,55 @@ async def stream(
             system_prompt: System prompt to provide context to the model.
 
         Yields:
-            Formatted message chunks from the model.
+            Model events.
+
+        Raises:
+            ContextWindowOverflowException: If the input exceeds the model's context window.
+            ModelThrottledException: If the model service is throttling requests.
+        """
+
+        def callback(event: Optional[StreamEvent] = None) -> None:
+            loop.call_soon_threadsafe(queue.put_nowait, event)
+            if event is None:
+                return
+
+            signal.wait()
+            signal.clear()
+
+        loop = asyncio.get_event_loop()
+        queue: asyncio.Queue[Optional[StreamEvent]] = asyncio.Queue()
+        signal = threading.Event()
+
+        thread = asyncio.to_thread(self._stream, callback, messages, tool_specs, system_prompt)
```

Comment:
If we make another edit, adding a comment indicating *why* we're doing this would be useful.  As-is, the context is mostly on GitHub/Commits, which is hard to pull out contextually as time goes on.